### ISSUE TITLE
Crash: SystemError: null argument to internal routine

### ISSUE THREAD

**Issue by NeilGirdhar** [PyCQA/pylint#5495]
### Bug description

Pylint crashes

### Command used

```shell
pylint cmm
```

### Pylint output

```shell
Traceback (most recent call last):
  File "shibokensupport/signature/loader.py", line 82, in make_helptext
  File "shibokensupport/signature/errorhandler.py", line 159, in make_helptext
SystemError: null argument to internal routine
Fatal Python error: handle_doc did not receive a result
Python runtime state: initialized

Current thread 0x00007fb4e2d38280 (most recent call first):
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/raw_building.py", line 244 in _base_class_object_build
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/raw_building.py", line 186 in object_build_class
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/raw_building.py", line 363 in object_build
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/raw_building.py", line 326 in inspect_build
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/builder.py", line 97 in module_build
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/manager.py", line 287 in ast_from_module
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/manager.py", line 190 in ast_from_module_name
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/nodes/scoped_nodes.py", line 735 in import_module
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/astroid/nodes/scoped_nodes.py", line 620 in getattr
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/checkers/variables.py", line 2168 in _check_module_attrs
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/checkers/variables.py", line 1339 in visit_importfrom
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 72 in walk
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 75 in walk
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1250 in _check_astroid_module
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1203 in check_astroid_module
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1069 in _check_file
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1023 in check_single_file_item
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/parallel.py", line 57 in _worker_check_single_file
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/pool.py", line 125 in worker
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/process.py", line 108 in run
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/process.py", line 315 in _bootstrap
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/popen_fork.py", line 71 in _launch
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/popen_fork.py", line 19 in __init__
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/context.py", line 277 in _Popen
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/process.py", line 121 in start
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/pool.py", line 326 in _repopulate_pool_static
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/pool.py", line 303 in _repopulate_pool
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/pool.py", line 212 in __init__
  File "/home/neil/.pyenv/versions/3.10.1/lib/python3.10/multiprocessing/context.py", line 119 in Pool
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/parallel.py", line 113 in check_parallel
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1000 in check
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/lint/run.py", line 398 in __init__
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/lib/python3.10/site-packages/pylint/__init__.py", line 24 in run_pylint
  File "/home/neil/.cache/pypoetry/virtualenvs/cmm-tspD8tmv-py3.10/bin/pylint", line 8 in <module>

Extension modules: lazy_object_proxy.cext, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg.lapack_lite, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, xxsubtype, shiboken6.Shiboken, PySide6.QtCore, PySide6.QtGui, PySide6.QtWidgets (total: 20)
```

### Expected behavior

no crash

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.1 (main, Dec  8 2021, 14:43:47) [GCC 9.3.0]
```

### OS / Environment

Ubuntu

**Comment by BoniLindsley** [comment]
I believe this is a bug from PySide6 version 6.2.2 on 2021-12-03. There is a patched version 6.2.2.1 out earlier today (2012-12-09). So upgrading or otherwise skipping 6.2.2 should fix this.

**Comment by BoniLindsley** [comment]
What I think is a minimal example:

```sh
echo import PySide6.QtCore > crasher.py
pylint --extension-pkg-allow-list=PySide6 crasher.py
```

The error changed after the upgrade. From `~/.cache/pylint/pylint-crash-2021-12-09-15.txt`:

> pylint crashed with a ``AttributeError`` and with the following stacktrace:
> ```
> Traceback (most recent call last):
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1034, in _check_files
>     self._check_file(get_ast, check_astroid_module, file)
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1069, in _check_file
>     check_astroid_module(ast_node)
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1203, in check_astroid_module
>     retval = self._check_astroid_module(
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1250, in _check_astroid_module
>     walker.walk(node)
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 75, in walk
>     self.walk(child)
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 72, in walk
>     callback(astroid)
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/checkers/variables.py", line 1320, in visit_import
>     self._check_module_attrs(node, module, parts[1:])
>   File "/home/boni/.local/lib/python3.9/site-packages/pylint/checkers/variables.py", line 2168, in _check_module_attrs
>     module = next(module.getattr(name)[0].infer())
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/nodes/scoped_nodes.py", line 620, in getattr
>     result = [self.import_module(name, relative_only=True)]
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/nodes/scoped_nodes.py", line 735, in import_module
>     return AstroidManager().ast_from_module_name(absmodname)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/manager.py", line 190, in ast_from_module_name
>     return self.ast_from_module(module, modname)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/manager.py", line 287, in ast_from_module
>     return AstroidBuilder(self).module_build(module, modname)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/builder.py", line 97, in module_build
>     node = self.inspect_build(module, modname=modname, path=path)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/raw_building.py", line 326, in inspect_build
>     self.object_build(node, module)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/raw_building.py", line 363, in object_build
>     class_node = object_build_class(node, member, name)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/raw_building.py", line 186, in object_build_class
>     return _base_class_object_build(node, member, basenames, localname=localname)
>   File "/home/boni/.local/lib/python3.9/site-packages/astroid/raw_building.py", line 244, in _base_class_object_build
>     member.__doc__,
> AttributeError: <class 'PySide6.QtCore.PyClassProperty'> object has no `__doc__` attribute
> ```

**Comment by NeilGirdhar** [comment]
@BoniLindsley Thank you for producing a MWE!

**Comment by BoniLindsley** [comment]
Likely related issue: <https://bugreports.qt.io/browse/PYSIDE-1727>

TLDR: Docstring generation was changed. A side effect was that a docstring does not always exist. This matches the "has no `__doc__` attribute" error.

**Comment by DanielNoord** [comment]
@BoniLindsley Would you be able to check out if the updated version as mentioned in that issue report fixes these issues? I believe it should. If so, we can close this issue.

**Comment by Pierre-Sassoulas** [comment]
Closing as won't fix, it is a crash in pylint, but the code it's ran on break some fundamental assertion about valid python code : namely that ``__doc__`` is defined. The maintainer refers to this here: https://bugreports.qt.io/browse/PYSIDE-1727?focusedCommentId=601513&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-601513.

**Comment by NeilGirdhar** [comment]
@DanielNoord The error persists on my end with `PySide6==6.2.2.1`.

**Comment by DanielNoord** [comment]
@NeilGirdhar Ah that is too bad! I don't fully understand how their repository works but I'm seeing commits referenced that were made after 9 December (the day of the latest release). Could it be that the fix has not been fully release yet?

**Comment by NeilGirdhar** [comment]
@DanielNoord I hope so.  Are they aware of the issue?

**Comment by DanielNoord** [comment]
I looked at their release notes and development notes ([here](https://wiki.qt.io/Qt_for_Python_Development_Notes)) and I think they consider it fixed (and probably included in the latest release).

Do you have some code that we can use to reproduce this? And could you share your pylint config (`pylinrc` or else)?

**Comment by NeilGirdhar** [comment]
Here's the relevant section of my pyproject.toml:
```toml
[tool.pylint.master]
jobs = 0
extension-pkg-whitelist = [
    'PySide6',
    'matplotlib.backends']

[tool.pylint.messages_control]
disable = [
    'abstract-method',
    'arguments-differ',
    'duplicate-code',
    'invalid-name',
    'invalid-unary-operand-type',
    'missing-docstring',
    'no-member',
    'no-self-use',
    'too-few-public-methods',
    'too-many-ancestors',
    'too-many-arguments',
    'too-many-branches',
    'too-many-instance-attributes',
    'too-many-locals',
    'too-many-return-statements',
    'too-many-statements',
    'unused-argument']
enable= [
    'useless-suppression',
    'use-symbolic-message-instead']

[tool.pylint.reports]
output-format = 'colorized'

[tool.pylint.similarities]
min-similarity-lines = 20
ignore-imports = 'yes'
```

**Comment by DanielNoord** [comment]
I have responded to the issue reported at PySide with some additional details. I'll let you know if I get a response!

**Comment by NeilGirdhar** [comment]
@DanielNoord Thanks a lot!!

**Comment by BoniLindsley** [comment]
I think they consider it fixed, for the next release. Likely not urgent enough for another patch release. Timeline was as follows:

- 2021-12-03: [Version 6.2.2](https://pypi.org/project/PySide6/6.2.2/) released on PyPI. It broke `help()` and `__doc__`.
- 2021-12-07: [Upstream issue](https://bugreports.qt.io/browse/PYSIDE-1727) about `help()` was created.
- 2021-12-08: [This issue](https://github.com/PyCQA/pylint/issues/5495) also about `help()` was created.
- 2021-12-09: [Version 6.2.2.1](https://pypi.org/project/PySide6/6.2.2.1/) released on PyPI. It fixed `help()` but not `__doc__`. Later the same day, they [commented](https://bugreports.qt.io/browse/PYSIDE-1727#comment-601513) that `__doc__` should be fixed in upstream source. No extra patch release was made, so only in source. 
- 2021-12-12: I [replied](https://bugreports.qt.io/browse/PYSIDE-1727#comment-601808) to their comment about `__doc__`, not knowing they had changed it. I guess I had skimmed that sentence. They did not seem happy in their reply about me bothering them.
- 2021-12-16: Version 6.2.2.1 said to be released on this day in [PySide6 development notes](https://wiki.qt.io/Qt_for_Python_Development_Notes#16._December_2021). [Wheels](https://download.qt.io/official_releases/QtForPython/pyside6/) in Qt's own download page was indeed modified on this day. However, the files have the same hashes as the 2012-12-09 release on PyPI. So not actually a new release, I think, but a rebuild with an announcement (unless something broke in their CD).

**Comment by DanielNoord** [comment]
Thanks @BoniLindsley! I updated my comment on their repository to reflect this timeline. Hopefully this fix gets included in a release soon!

**Comment by DanielNoord** [comment]
This fix was indeed not included in the hotfix. There are currently no plans for a new hotfix, but anybody affected by this could respond to their issue and see if that's a possibility. 

The next normal release is expected mid January, so the issue should be resolved by then. I'm closing this as there is not much we can do at our end and the issue has been resolved in an upcoming release.

**Comment by NeilGirdhar** [comment]
> The next normal release is expected mid January, so the issue should be resolved by then. I'm closing this as there is not much we can do at our end and the issue has been resolved in an upcoming release.

If this happens again, I suggest at least catching the exception and maybe outputting a single line error.  That would be much more gracefull than stack traces.

**Comment by DanielNoord** [comment]
Normally I think all exceptions are automatically turned into an issue template for the user to easily report the issue here. It might be that that isn't working when we're using multiple threads and running `pylint` in parallel.

@Pierre-Sassoulas Do you know something about this?

**Comment by Pierre-Sassoulas** [comment]
I think the template should be provided when the issue happens in astroid, (because astroid is hard to test as we'd need to provide all possible code construct which is not reasonable. If something crash in pylint it means we did not test properly). It looks like there's a problem in astroid and the template is not created so I opened https://github.com/PyCQA/pylint/issues/5668 to follow up.
